test/suite/test_truncate11.py

```diff
@@ -0,0 +1,117 @@
+
+import threading, time, wttest
+from wtdataset import simple_key, simple_value
+from wtscenario import make_scenarios
+from wiredtiger import stat
+from wtthread import checkpoint_thread
+
+class test_truncate11(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=50MB,statistics=(all),statistics_log=(json,on_close,wait=1),timing_stress_for_test=[checkpoint_slow]'
+
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('column_fix', dict(key_format='r', value_format='8t')),
+        ('row_integer', dict(key_format='i', value_format='S')),
+    ]
+
+    scenarios = make_scenarios(format_values)
+
+    def test_truncate11(self):
+        # Create a large table with lots of pages.
+        uri = "table:test_truncate11"
+        format = 'key_format={},value_format=S'.format(self.key_format)
+        self.session.create(uri, 'allocation_size=512,leaf_page_max=512,' + format)
+
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, 80000):
+            cursor[simple_key(cursor, i)] = simple_value(cursor, i)
+        cursor.close()
+
+        valuea = "aaaaa" * 100
+
+        # Force to disk.
+        self.reopen_conn()
+
+        # Set the oldest timestamp and the stable timestamp.
+        self.conn.set_timestamp('oldest_timestamp='+ self.timestamp_str(100))
+        self.conn.set_timestamp('stable_timestamp='+ self.timestamp_str(100))
+
+        # Update a large number of records.
+        cursor = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        for i in range(1, 10):
+            cursor[i] = valuea
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(120))
+        cursor.close()
+
+        # Create a checkpoint thread
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
+        try:
+            ckpt.start()
+
+            # Wait for checkpoint to start before committing last transaction.
+            ckpt_started = 0
+            while not ckpt_started:
+                time.sleep(1)
+                stat_cursor = self.session.open_cursor('statistics:', None, None)
+                ckpt_started = stat_cursor[stat.conn.txn_checkpoint_running][2]
+                stat_cursor.close()
+
+            # Start a transaction.
+            self.session.begin_transaction()
+
+            # Truncate a chunk.
+            c1 = self.session.open_cursor(uri, None)
+            c1.set_key(simple_key(c1, 20000))
+            c2 = self.session.open_cursor(uri, None)
+            c2.set_key(simple_key(c1, 40000))
+
+            self.session.truncate(None, c1, c2, None)
+
+            # Commit the transaction.
+            self.session.timestamp_transaction('commit_timestamp=' + self.timestamp_str(150))
+            self.session.commit_transaction()
+
+        finally:
+            done.set()
+            ckpt.join()
+
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
```

Comment:
Could you extend this test to open a checkpoint cursor, and verify that the checkpoint does include content truncated before it started, and doesn't include content truncated after it started (but ideally before checkpoint finishes).

Reply:
FWIW: I'm conscious that we will want to backport this to a version of WiredTiger that doesn't support checkpoint cursors. It would be reasonable to add an extended version of this test as part of another ticket, but I'd like to at least know the validation has happened before this change has been merged.

Reply:
Glad it works -- some things at the intersection of checkpoint cursors and fast-truncate don't until WT-9252 lands again  :-|

Reply:
As part of this test, I didn't do explicit eviction as it doesn't work until WT-9252. The attached test can be enhanced further once WT-9252 lands again.